webpack/translations-plugin.js

```diff
@@ -0,0 +1,68 @@
+const path = require('path')
+const fs = require('fs')
+const JS_INDENT = 2
+const FILENAME = 'en'
+
+const marketplaceKeys = [
+  'name',
+  'description',
+  'short_description',
+  'long_description',
+  'installation_instructions',
+  'parameters'
+]
+
+class TranslationsPlugin {
+  constructor (options) {
+    this.options = options
+  }
+
+  // Defines `apply` method in it's prototype.
+  apply (compiler) {
+    // Specifies webpack's event hook to attach itself.
+    compiler.hooks.emit.tapAsync('TranslationsPlugin', (compilation, callback) => {
+      Object.assign(
+        compilation.assets,
+        buildMarketplaceTranslationFile('en.json', this.options.path)
+      )
+      callback()
+    })
+  }
+}
+
+function buildMarketplaceTranslationFile (filename, filepath) {
+  let translationsInput
+  try {
+    translationsInput = JSON.parse(fs.readFileSync(path.resolve(filepath, filename)))
+  } catch (err) {
+    console.error(err)
+    process.exit(1)
+  }
+
+  const translationsPath = `../translations/${filename}`
+  const marketplaceTranslations = extractMarketplaceTranslation(translationsInput, filename)
+
+  return {
+    [translationsPath]: {
```

Comment:
:lolwut: ?

Reply:
https://speakerdeck.com/erikrose/constructive-code-review?slide=16

Reply:
```
// Computed property names (ES2015)
var prop = 'foo';
var o = {
  [prop]: 'hey',
  ['b' + 'ar']: 'there'
};
```

Reply:
thanks for that